### PR TITLE
chore: drop Node.js v10

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [x] This is a **metadata update**

### For Bugs and Features; did you add new tests?
no

### Motivation / Use-Case
Node.js v10 is reaching EOL soon, this will drop it from CI

### Breaking Changes
Node v10 is no longer officially supported

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "client"
   ],
   "engines": {
-    "node": ">= 10.13.0"
+    "node": ">= 12.x.x"
```
Comment (alexander-akait) [inline package.json:14]: Please use LTS here https://nodejs.org/en/blog/release/v12.13.0/

Comment (anshumanv) [reply package.json]: done

Comment (snitin315) [reply package.json]: I think we should add `npm` here too to avoid oscillating `package-lock.json` between v1 and v2

Comment (anshumanv) [reply package.json]: yep lets add it, huge lock diffs otherwise

**Review by alexander-akait — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Pull request merged**